ethcore/call-contract/src/call_contract.rs

```diff
@@ -16,14 +16,70 @@
 
 //! Provides CallContract and RegistryInfo traits
 
+use std::fmt;
 use bytes::Bytes;
-use ethereum_types::Address;
+use derive_builder::Builder;
+use ethereum_types::{Address, U256};
 use types::ids::BlockId;
 
+#[derive(Debug)]
+/// Transaction call error
+pub enum CallError {
+	/// Call reverted
+	Reverted(String),
+	/// Other error
+	Other(String),
+}
```

Comment:
api consumer doesn't doesn't care whether the call is reverted has been reverted or not, which, imo makes this new data structure redundant